### PR TITLE
[FW] l10n_it_delivery_note: multipli fw-ports e il loro adattamento

### PR DESCRIPTION
**Pull request by PicchiSeba**
Fw-port di diversi commit per `l10n_it_delivery_note` necessari per l'aggiornamernto alla 16.0 di https://github.com/OCA/l10n-italy/issues/3938

Rispettivamente corrispondono alle seguenti PR:
- https://github.com/OCA/l10n-italy/pull/2346
- https://github.com/OCA/l10n-italy/pull/3303 
- https://github.com/OCA/l10n-italy/pull/3413
- https://github.com/OCA/l10n-italy/pull/3834
- https://github.com/OCA/l10n-italy/pull/3838
- https://github.com/OCA/l10n-italy/pull/3936

Sono consapevole che ci siano molti cambiamenti in una sola PR, se lo ritenete necessario posso fare ogni porting in una PR a sé stante, chiaramente ci vorrà più tempo

### CODE REVIEW

**Comment by PicchiSeba** [comment]
Il fallimento dei test non mi sembra collegato ai cambiamenti che ho fatto in questa PR. Sembra più un problema relativo al numero di sessioni che possono venire aperte contemporaneamente

**Comment by PicchiSeba** [comment]
> 2024-03-07 15:34:32,123 485 WARNING odoo odoo.addons.currency_rate_update.models.res_currency_rate_provider: Currency Rate Provider "Bank of Italy" failed to obtain data since 2023-03-07 until 2024-03-07 
Traceback (most recent call last):
  File "/opt/odoo-venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/opt/odoo-venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 445, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/opt/odoo-venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 440, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.10/http/client.py", line 1375, in getresponse
    response.begin()
  File "/usr/lib/python3.10/http/client.py", line 318, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.10/http/client.py", line 287, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response

> During handling of the above exception, another exception occurred:

> Traceback (most recent call last):
  File "/opt/odoo-venv/lib/python3.10/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/opt/odoo-venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 755, in urlopen
    retries = retries.increment(
  File "/opt/odoo-venv/lib/python3.10/site-packages/urllib3/util/retry.py", line 532, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/opt/odoo-venv/lib/python3.10/site-packages/urllib3/packages/six.py", line 769, in reraise
    raise value.with_traceback(tb)
  File "/opt/odoo-venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/opt/odoo-venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 445, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/opt/odoo-venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 440, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.10/http/client.py", line 1375, in getresponse
    response.begin()
  File "/usr/lib/python3.10/http/client.py", line 318, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.10/http/client.py", line 287, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

> During handling of the above exception, another exception occurred:

> Traceback (most recent call last):
  File "/opt/odoo-venv/lib/python3.10/site-packages/odoo/addons/currency_rate_update/models/res_currency_rate_provider.py", line 132, in _update
    data = provider._obtain_rates(
  File "/__w/l10n-italy/l10n-italy/currency_rate_update_boi/models/res_currency_rate_provider_BOI.py", line 275, in _obtain_rates
    response = requests.get(
  File "/opt/odoo-venv/lib/python3.10/site-packages/requests/api.py", line 76, in get
    return request('get', url, params=params, **kwargs)
  File "/opt/odoo-venv/lib/python3.10/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/opt/odoo-venv/lib/python3.10/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/opt/odoo-venv/lib/python3.10/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/opt/odoo-venv/lib/python3.10/site-packages/requests/adapters.py", line 498, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2024-03-07 15:34:32,142 485 INFO odoo odoo.addons.currency_rate_update_boi.tests.test_currency_rate_update_boi: ====================================================================== 
2024-03-07 15:34:32,142 485 ERROR odoo odoo.addons.currency_rate_update_boi.tests.test_currency_rate_update_boi: FAIL: TestCurrencyRateUpdate.test_foreign_base_currency
Traceback (most recent call last):
  File "/__w/l10n-italy/l10n-italy/currency_rate_update_boi/tests/test_currency_rate_update_boi.py", line 157, in test_foreign_base_currency
    self.test_update_BOI_year()
  File "/__w/l10n-italy/l10n-italy/currency_rate_update_boi/tests/test_currency_rate_update_boi.py", line 95, in test_update_BOI_year
    self.assertTrue(rates)
AssertionError: res.currency.rate() is not true

**Review by Borruso — APPROVED** [review]
LGTM

File: l10n_it_delivery_note/report/report_delivery_note.xml [hunk 1]
```diff
@@ -269,7 +306,7 @@
                                 style="font-size:10px;"
                             >Gross Weight</div>
                             <div
-                                class="text-end"
```
Comment (Borruso) [inline l10n_it_delivery_note/report/report_delivery_note.xml:272]: per la v16 il la classe è cambiata va bene `text-end`

**Review by Borruso — CHANGES_REQUESTED** [review]

**Review by Borruso — APPROVED** [review]
LGTM

**Review by aleuffre — APPROVED** [review]
Code review

**Comment by francesco-ooops** [comment]
@tafaRU puoi provvedere tu?

**Comment by aleuffre** [comment]
No, c'è un file di troppo, pandoc, da rimuovere.

**Pull request merged**